arch/riscv/core/reset.S

```diff
@@ -21,6 +22,8 @@ GTEXT(riscv_cpu_wake_flag)
 GTEXT(riscv_cpu_sp)
 GTEXT(z_riscv_secondary_cpu_init)
 
+GDATA(_non_z_stacks)
```

Comment:
`non_z`?

Reply:
"Non Zephyr" i.e. stacks for cpus not participating in the zephyr cluster.

Reply:
And why do we need this in the general arch code? I mean, this looks to me as something mpfs_icicle-specific, or at least specific to your use case.

Could you describe in more detail (and ideally - add such description to the code itself) why and how you use this?

Reply:
The use case for SMP to date has been rather simplistic with a simple CPU setup that assumes Zephyr has total access to all the CPUs on the system (or at least the first n...). Myself and the team I work with have been involved with low level code development on the MPFS and initially the FU540 for some time now and have seen quite a number of use cases for multi CPU SOCs being looked at by clients. Real world examples have included simple SMP setups, AMP setups involving different combinations of RTOS/Linux/bare metal and other ways to make use of the available cores. With this in mind, I looked at how an RTOS like Zephyr could fit into these type of systems. That's why I settled on the scheme in this code to try and allow the Zephyr component of such systems share the CPU set in a flexible fashion. I also needed to be able to support non-homogeneous cpu setups as MPFS and FU540 are such systems and I am aware of some others on the horizon. Part of the development also looks at operation of Zephyr on real SMP hardware which does not always follow the operation in systems such as QEMU or Renode as they tend to make some assumptions about the low level operation of the CPUs which do not hold true in real life... So, whilst the development has been carried out on MPFS, issues I am dealing with are more generally applicable.